### PR TITLE
[Feature] Broker support COS posix bucket

### LINKED ISSUE #46020: Support Tencent Cloud COS posix bucket
- [x] https://github.com/StarRocks/starrocks/pull/46820
- [x] https://github.com/StarRocks/starrocks/pull/46821
- [x] https://github.com/StarRocks/starrocks/pull/46597
- [x] https://github.com/StarRocks/starrocks/pull/46090
- [x] https://github.com/StarRocks/starrocks/pull/46089
- [x] https://github.com/StarRocks/starrocks/pull/46088
- [x] https://github.com/StarRocks/starrocks/pull/42638

## Feature request

**Is your feature request related to a problem? Please describe.**

https://cloud.tencent.com/document/product/436/56971
The Tencent Cloud release a new feature of COS.
**Describe the solution you'd like**

Support the read/write of COS posix bucket.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by wyb**
## What I'm doing:

This PR adds the support of COS posix bucket.
With broker, you can use Broker Load to load data from COS posix bucket, and use SELECT INTO OUTFILE to export data to COS posix bucket.
This is maily inspired by https://cloud.tencent.com/document/product/436/82208.
```
LOAD LABEL db0.label3 (     DATA INFILE("cosn://xxx/some.parquet")     INTO TABLE tbl_student     FORMAT AS "parquet"     (name, age)  ) WITH BROKER  srbroker ( 
"fs.cosn.bucket.endpoint_suffix" = "cos.ap-guangzhou.myqcloud.com",
"fs.cosn.userinfo.region"="ap-guangzhou",
"qcloud.object.storage.ranger.service.address"="10.0.0.26:9999",
"hadoop.security.authentication"="kerberos",
"fs.ofs.tmp.cache.dir"="/tmp",
"fs.cosn.trsf.fs.ofs.user.appid"="12345",
"qcloud.object.storage.kerberos.principal"="hadoop/10.0.0.26@EMR-ABC",
"kerberos_principal"="hadoop/10.0.0.26@EMR-ABC", 
"kerberos_keytab"="/var/krb5kdc/emr.keytab"
);
```
```
SELECT * FROM tbl_student INTO OUTFILE "cosn://some_bucket/parquet/" FORMAT AS PARQUET PROPERTIES (
"broker.name"="srbroker",
"fs.cosn.bucket.endpoint_suffix" = "cos.ap-guangzhou.myqcloud.com",
"fs.cosn.userinfo.region"="ap-guangzhou",
"qcloud.object.storage.ranger.service.address"="10.0.0.26:9999",
"hadoop.security.authentication"="kerberos",
"fs.ofs.tmp.cache.dir"="/tmp",
"fs.cosn.trsf.fs.ofs.user.appid"="12345",
"qcloud.object.storage.kerberos.principal"="hadoop/10.0.0.26@EMR-ABC",
"kerberos_principal"="hadoop/10.0.0.26@EMR-ABC", 
"kerberos_keytab"="/var/krb5kdc/emr.keytab"
);
```
Fixes https://github.com/StarRocks/starrocks/issues/46020
Fixes compatibility issue https://github.com/StarRocks/starrocks/pull/46425

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fs_brokers/apache_hdfs_broker/build.sh [hunk 1]
```diff
@@ -25,14 +25,17 @@ export STARROCKS_HOME=${ROOT}/../..
...
+export HADOOP_COS_HOME=$ROOT/src/hadoop-cos-shaded/
 
 # prepare thrift
-mkdir -p ${BROKER_HOME}/src/main/resources/thrift
-mkdir -p ${BROKER_HOME}/src/main/thrift
-
-cp ${BROKER_HOME}/../../gensrc/thrift/FileBrokerService.thrift ${BROKER_HOME}/src/main/resources/thrift/
+mkdir -p ${BROKER_CORE_HOME}/src/main/resources/thrift
+mkdir -p ${BROKER_CORE_HOME}/src/main/thrift
+cp ${BROKER_HOME}/../../gensrc/thrift/FileBrokerService.thrift ${BROKER_CORE_HOME}/src/main/resources/thrift/
```
Comment (kevincai) [inline fs_brokers/apache_hdfs_broker/build.sh:36]: can we makek this a symbolic link?

**Review by kevincai — DISMISSED** [review]

File: fs_brokers/apache_hdfs_broker/build.sh (hunk 1 above)
Comment (wyb) [reply fs_brokers/apache_hdfs_broker/build.sh]: done

**Review by kevincai — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wyb** [comment]
@Mergifyio backport branch-3.1

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.1

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.2

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.3

**Pull request merged**